scalopus_general/include/scalopus_general/map_tracker.h

```diff
@@ -61,12 +61,22 @@ class MapTracker
    * @brief Function to retrieve the mapping between entry and exit ids and provided names.
    * @return Map that holds trace point id - trace name mapping.
    */
-  std::map<Key, Value> getMap() const
+  std::unordered_map<Key, Value> getMap() const
   {
     std::shared_lock<decltype(mutex_)> lock(mutex_);
     return mapping_;
   }
 
+  /**
+   * @brief Retrieve a key from the map, the caller is responsible for ensuring the key exists.
+   * @return The value stored for this key.
+   */
+  Value getKey(const Key& k) const
```

Comment:
Technically, it should be `getValue`. In the doxygen comment, mention that it's returned by copy intentionally, so the user doesn't have to worry about thread safety.

Reply:
Hah, I think I wanted to type `getByKey` but then lost a word. I like `getValue` better anyway. Will improve doxygen. Thanks for catching this.